SUI/SUI/ActionSheetView.swift

```diff
@@ -0,0 +1,184 @@
+                .fill(.black)
+                .frame(width: 80, height: 3)
+                .padding(.top)
+            HStack(spacing: 60) {
+                titleTextView
+                onButtonView
+            }
+            .padding(.bottom, 20)
+            HStack(spacing: 30) {
+                ColorPicker(Constants.emptyString, selection: $climateViewModel.selectedColor)
+                Spacer()
+                decreaseButtonView
+                celsusTextView
+                increaseButtonView
+                Spacer()
+                Image(Constants.carDoorImageName)
+            }
+            .padding(.horizontal, 40)
+            .padding(.bottom, 20)
+            bottomTextView
+            Spacer()
+        }
+        .frame(width: UIScreen.main.bounds.width, height: UIScreen.main.bounds.height / 2)
+        .background(
+            .ultraThinMaterial, in:
+            RoundedRectangle(cornerRadius: 40)
+        )
+        .ignoresSafeArea(.all, edges: .bottom)
+        .offset(y: UIScreen.main.bounds.height / 2 + 120)
+        .offset(y: currentMenuOffsetY)
+        .gesture(dragGesture)
+    }
+
+    // MARK: - Private properties
+
+    @EnvironmentObject private var climateViewModel: ClimateViewModel
+
+    @GestureState private var gestureOffset = CGSize.zero
+
+    @State private var currentMenuOffsetY: CGFloat = 0.0
+    @State private var lastMenuOffsetY: CGFloat = 0.0
+
+    private var titleTextView: some View {
+        VStack(alignment: .leading, spacing: 5) {
+            Text(Constants.titleText)
+                .font(.system(size: 20, weight: .bold, design: .default))
+                .foregroundColor(.white)
+
+            Text(Constants.descriptionText)
+                .lineLimit(2)
+                .frame(width: 200, alignment: .leading)
+                .foregroundColor(.gray)
+        }
+        .padding(.leading)
+    }
+
+    private var bottomTextView: some View {
+        HStack {
+            Text(Constants.onString)
+            Spacer()
+            Text(Constants.ventString)
+        }
+        .padding(.horizontal, 40)
+    }
+
+    private var celsusTextView: some View {
+        Text("\(climateViewModel.currentCelsus)\(Constants.celsusString)")
+            .font(.system(size: 30, weight: .medium, design: .default))
+            .frame(width: 50)
+    }
+
+    private var dragGesture: some Gesture {
+        DragGesture()
+            .updating($gestureOffset) { value, state, _ in
+                state = value.translation
+                onChangeMenuOffset()
+            }
+            .onEnded { _ in
+                let maxHeight = UIScreen.main.bounds.height / 6
+                withAnimation {
+                    if -currentMenuOffsetY > maxHeight / 2 {
+                        currentMenuOffsetY = -maxHeight
+                    } else {
+                        currentMenuOffsetY = 0
+                    }
+                    lastMenuOffsetY = currentMenuOffsetY
```

Comment:
```suggestion
                let maxHeight = UIScreen.main.bounds.height / 6
                withAnimation {
                    if -currentMenuOffsetY > maxHeight / 2 {
                        currentMenuOffsetY = -maxHeight
                    } else {
                        currentMenuOffsetY = 0
                    }
                    lastMenuOffsetY = currentMenuOffsetY
```
Мне нравится - универсально, я захардкодил

Reply:
это из видео Жени